common/utils/src/main/resources/error/error-classes.json

```diff
@@ -3213,6 +3225,12 @@
     ],
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_DATA_TYPE_FOR_SIZE_FUNCTION" : {
```

Comment:
I think we can remove this error class because:
```scala
  override def inputTypes: Seq[AbstractDataType] = Seq(TypeCollection(ArrayType, MapType))
```
and the case must be unreachable:
```scala
      case other => throw QueryExecutionErrors.unsupportedOperandTypeForSizeFunctionError(other)
```
@dengziming Please, replace it by an internal error.